### PR TITLE
Bugfix FXIOS-8351 Fix URL bar update during navigation

### PR DESCRIPTION
**Pull request by mattreaganmozilla**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8351)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18485)

## :bulb: Description

Fixes an issue where WebEngine delegates aren't notified of a URL change.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by mattreaganmozilla** [comment]
@lmarceau LMK if you have any thoughts or want to sync up further. This was the check (in `handleURLChange`) which was preventing the delegate from being notified (to update the location bar):

```
        // To prevent spoofing, only change the URL immediately if the new URL is on
        // the same origin as the current URL. Otherwise, do nothing and wait for
        // didCommitNavigation to confirm the page load.
guard sessionData.url?.origin == webView.url?.origin else { return }`
```

I considered a couple approaches and opted for the simplest; my concern with this however is that when navigating within the same origin we'll be running the code twice that notifies delegates & fetches metadata.

I thought about setting a special state in the case where we detect a URL change but the origins don't match, so that we can check that and only call into `commitURLChange` in that case to avoid this 2x delegate callback problem. But there may be scenarios with redirects where that could be problematic also.

File: BrowserKit/Sources/WebEngine/WKWebview/WKEngineSession.swift [hunk 1]
```diff
@@ -371,6 +376,7 @@ class WKEngineSession: NSObject,
     func webView(_ webView: WKWebView,
```
Comment (lmarceau) [inline BrowserKit/Sources/WebEngine/WKWebview/WKEngineSession.swift:376]: Seems great! I would add a follow up task to add unit test for this use case when we have https://mozilla-hub.atlassian.net/browse/FXIOS-8477 as well

**Review by lmarceau — COMMENTED** [review]
Seems great to me! I think we can start with this approach and monitor in case it needs further improvement down the line (like too many updates to metadata as you said or else).

**Review by lmarceau — APPROVED** [review]

File: BrowserKit/Sources/WebEngine/WKWebview/WKEngineSession.swift (hunk 1 above)
Comment (mattreaganmozilla) [reply BrowserKit/Sources/WebEngine/WKWebview/WKEngineSession.swift:376]: Good point, I added a task; also added an inline TODO just to make sure this stays on the radar

**Pull request merged**